.github/workflows/rust.yml

```
@@ -108,14 +108,37 @@ jobs:
       run: |
         cargo install --path cargo-apk
         rustup target add x86_64-linux-android
+    - name: AVD cache
+      uses: actions/cache@v2
+      id: avd-cache
+      with:
+        path: |
+          ~/.android/avd/*
+          ~/.android/adb*
+        # One cache for 3 apk files
+        key: avd-emul
+    - name: create AVD and generate snapshot for caching
+      if: steps.avd-cache.outputs.cache-hit != 'true'
+      uses: reactivecircus/android-emulator-runner@v2.19.1
+      with:
+        api-level: 29
+        arch: x86_64
+        profile: Nexus 6
+        target: default
+        force-avd-creation: true
+        emulator-options: -no-window -gpu swiftshader_indirect -noaudio -no-boot-anim -camera-back none
+        disable-animations: false
```

Let's keep this as similar as possible to the actual use:
```suggestion
        api-level: 29
        arch: x86_64
        # the `googleapis` emulator target is considerably slower on CI.
        target: default
        profile: Nexus 6
        disable-animations: true # Still not sure
        emulator-options: -no-window -gpu swiftshader_indirect -noaudio -no-boot-anim -camera-back none
        force-avd-creation: true
```